src/hooks/useOperationModal/Modal/SupplyForm/SubmitSection/index.tsx

```diff
@@ -72,12 +84,36 @@ export const SubmitSection: React.FC<SubmitSectionProps> = ({
       });
     }
 
+    if (!isFormSubmitting && formError === 'HIGHER_THAN_WALLET_SPENDING_LIMIT') {
+      return t('operationModal.supply.submitButtonLabel.amountHigherThanWalletWalletSpendingLimit');
+    }
+
+    if (!isFormSubmitting && formError === 'PRICE_IMPACT_TOO_HIGH') {
```

Comment:
Some ideas to improve this 
- Use yup or zod to handle form validation and return the error message. This removes the need for all the error condiationals
- This could be abstracted to a function that does a single check for `isFormSubmitting`

Reply:
Agreed, that would make things cleaner. I've created a ticket (VEN-1732) for this task.